miner/builder.go

```diff
@@ -176,6 +179,53 @@ func (b *Builder) GetBalance(addr common.Address) *big.Int {
 	return b.env.state.GetBalance(addr).ToBig()
 }
 
+type ChainContextDummy struct {
+}
+
+func (c *ChainContextDummy) Engine() consensus.Engine {
+	panic("not implemented")
```

Comment:
Can you return a log.Error in this situation instead of panic? Same for GetHeader.

Reply:
Implemented these changes in the latest commit